gbasis/spherical.py

```diff
@@ -0,0 +1,286 @@
+    if np.abs(mag) > angmom:
+        raise ValueError("The magnetic quantum number must be between -(angmom) and angmom.")
+    if not isinstance(i, int):
+        raise TypeError("Index i must be an integer")
+    if not isinstance(j, int):
+        raise TypeError("Index j must be an integer")
+    if isinstance(k, int):
+        k = float(k)
+    if not isinstance(k, float):
+        raise TypeError("Index k must be a float.")
+    if k != int(k) and mag >= 0:
+        raise ValueError("Index k must be an integer for non-negative magnetic quantum numbers.")
+    if k != int(k) + 0.5 and mag < 0:
+        raise ValueError("Index k must be a half integer for negative magnetic quantum numbers.")
+
+    if mag < 0:
+        return np.real(
+            (complex(-1)) ** (i + k - shift_factor(mag))
+            * (1 / 4) ** i
+            * comb(angmom, i)
+            * comb(angmom - i, np.abs(mag) + i)
+            * comb(i, j)
+            * comb(np.abs(mag), 2 * k)
+        )
+    return (
+        (-1) ** (i + k - shift_factor(mag))
+        * (1 / 4) ** i
+        * comb(angmom, i)
+        * comb(angmom - i, np.abs(mag) + i)
+        * comb(i, j)
+        * comb(np.abs(mag), 2 * k)
+    )
+
+
+def harmonic_norm(angmom, mag):
+    """Calculate the normalization constant of a real solid harmonic.
+
+        :math: `N^S_angmom,m = 1/(2^abs{m} * angmom!) * sqrt{2 * (angmom + abs{m})!
+        * (angmom - abs{m})! / 2^del{0, m})}`,
+        where :math: `del{0, m}` is the Kronecker delta of 0 and m.
+
+    Parameters
+    ----------
+    angmom : int
+        The angular momentum of the basis function(s).
+
+    mag : int
+        The magnetic quantum number(s) of the basis function(s).
+
+    Returns
+    -------
+    norm : float
+        The normalization constant of the real solid harmonic.
+
+    Raises
+    ------
+    TypeError
+        If angmom is not an integer.
+        If mag is not an integer.
+    ValueError
+        If angmom is negative.
+        If mag has a greater magnitude than angmom.
+
+    """
+    if not isinstance(angmom, int):
+        raise TypeError("Angular momentum must be an integer.")
+    if angmom < 0:
+        raise ValueError("Angular momentum must be a non-negative integer.")
+    if not isinstance(mag, int):
+        raise TypeError("The magnetic quantum number must be an integer.")
+    if np.abs(mag) > angmom:
+        raise ValueError("The magnetic quantum number must be between -(angmom) and angmom.")
+
+    return (1 / (2 ** np.abs(mag) * factorial(angmom))) * np.sqrt(
+        (2 * factorial(angmom + np.abs(mag)) * factorial(angmom - np.abs(mag)))
+        / (2 ** int(mag == 0))
+    )
+
+
+def real_solid_harmonic(angmom, mag):
+    r"""Calculate a real solid harmonic.
+
+    ..math::
+
+        S_angmom,mag = N^S_angmom,mag \sum_i=0^[(angmom-abs{mag})/2] \sum_j=0^i
+        \sum_k=shift_factor^([(abs{mag}-1)/2] + 0.5) C^angmom,m_i,j,k * x^a_x * y^a_y * z^a_z,
```

Comment:
I think you need a blank line after the math equation to delimit the latex environment